scripts/cert_gen.sh

```diff
@@ -0,0 +1,402 @@
+#!/bin/bash
+
+## Copyright (c) Microsoft. All rights reserved.
+## Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+###############################################################################
+# This script demonstrates creating X.509 certificates for an Azure IoT Hub
+# CA Cert deployment.
+# This script is used to create the python build environment
+# on which our e2e tests run.
+# Currently the root cert will have a duration of 1 year.
+# All other certificates created from this root certificate
+# gets deleted after the tests.
+# NOTE : In case of 401 error during running DPS E2E tests for certificates
+# the certificates produced by this has expired. And the build environment
+# needs to run again.
+#
+# These certs MUST NOT be used in production.  It is expected that production
+# certificates will be created using a company's proper secure signing process.
+# These certs are intended only to help demonstrate and prototype CA certs.
+###############################################################################
+
+root_ca_dir="."
+home_dir="."
+algorithm="genrsa"
+COUNTRY="US"
+STATE="WA"
+LOCALITY="Redmond"
+ORGANIZATION_NAME="My Organization"
+root_ca_password="1234"
+key_bits_length="4096"
+days_till_expire=365
+ca_chain_prefix="azure-iot-test-only.chain.ca"
+intermediate_ca_dir="."
+openssl_root_config_file="./openssl_root_ca.cnf"
+openssl_intermediate_config_file="./openssl_device_intermediate_ca.cnf"
+intermediate_ca_password="1234"
+root_ca_prefix="azure-iot-test-only.root.ca"
+intermediate_ca_prefix="azure-iot-test-only.intermediate"
+
+function makeCNsubject()
+{
+    local result="/CN=${1}"
+    case $OSTYPE in
+        msys|win32) result="/${result}"
+    esac
+    echo "$result"
+}
+
+function warn_certs_not_for_production()
+{
+    tput smso
+    tput setaf 3
+    echo "Certs generated by this script are not for production (e.g. they have hard-coded passwords of '1234'."
+    echo "This script is only to help you understand Azure IoT Hub CA Certificates."
+    echo "Use your official, secure mechanisms for this cert generation."
+    echo "Also note that these certs will expire in ${days_till_expire} days."
+    tput sgr0
+}
+
+function generate_root_ca()
+{
+    local common_name="Azure IoT Hub CA Cert Test Only"
+    local password_cmd=" -aes256 -passout pass:${root_ca_password} "
```

Comment:
Does this line mean you have to have a password on your cert.  If so do we want that?

Reply:
Right now it is very tightly coupled with original script. I did not change it. As of now I would let it be. I want to try something later.

---
In reply to: [396884105](https://github.com/Azure/azure-iot-sdk-python/pull/508#discussion_r396884105) [](ancestors = 396884105)